server/src/test/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStreamTests.java

```diff
@@ -79,6 +83,42 @@ protected InputStream openSlice(int slice) throws IOException {
         }
     }
 
+    public void testSkip() throws IOException {
+        final int slices = randomIntBetween(1, 20);
+        final var bytes = randomByteArrayOfLength(randomIntBetween(1000, 10000));
+        final int sliceSize = bytes.length / slices;
+
+        final var streamsOpened = new ArrayList<CheckClosedInputStream>();
+        SlicedInputStream input = new SlicedInputStream(slices) {
+            @Override
+            protected InputStream openSlice(int slice) throws IOException {
+                final int sliceOffset = slice * sliceSize;
+                final int length = slice == slices - 1 ? bytes.length - sliceOffset : sliceSize;
+                final var stream = new CheckClosedInputStream(new ByteArrayInputStream(bytes, sliceOffset, length));
+                streamsOpened.add(stream);
+                return stream;
+            }
+        };
+
+        // Skip up to a random point
+        final int skip = randomIntBetween(0, bytes.length);
+        input.skipNBytes(skip);
+
+        // Read all remaining bytes, which should be the bytes from skip up to the end
+        final int remainingBytes = bytes.length - skip;
+        if (remainingBytes > 0) {
+            final var remainingBytesRead = new byte[remainingBytes];
+            input.readNBytes(remainingBytesRead, 0, remainingBytes);
+            final var expectedRemainingBytes = new ByteArrayInputStream(bytes, skip, remainingBytes).readAllBytes();
```

Comment:
I find this simpler:

```suggestion
            final var expectedRemainingBytes = Arrays.copyOfRange(bytes, skip, bytes.length);
```